am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/RevokeAccessIntegrationTest.java

```diff
@@ -27,18 +39,109 @@ void setupTest() {
         resourceId = UUID.randomUUID().toString();
     }
 
+
     @Test
-    void whenRevokingResourceAccessResourceAccessRemovedFromDatabase() {
+    void whenRevokingResourceAccessShouldRemoveFromDatabase() {
         ams.grantExplicitResourceAccess(createGrantForWholeDocument(resourceId, READ_PERMISSION));
         ams.revokeResourceAccess(createMetadata(resourceId));
 
         assertThat(databaseHelper.countExplicitPermissions(resourceId)).isEqualTo(0);
     }
 
+    @Test
+    void whenRevokingResourceAccessOnRootShouldRemoveFromDatabase() {
+        ams.grantExplicitResourceAccess(createGrantForWholeDocument(resourceId, READ_PERMISSION));
+        revokeResourceAccess("");
+
+        assertThat(databaseHelper.countExplicitPermissions(resourceId)).isEqualTo(0);
+    }
+
+    @Test
+    void whenRevokingResourceAccessOnSingleNestedAttributeShouldRemoveFromDatabase() {
+        grantExplicitResourceAccess(resourceId, "/test");
```

Comment:
To be honest I would prefer to operate on more real examples e.g.

```
{
  "claimant": {
    "name": "John",
    "age": 21,
    "address": {
      "city": "London",
      "postcode": "SE1"
    }
  },
  "defendant": {
    "name": "Marry",
    "address": {
      "city": "Swansea",
      "postcode": "SA1"
    }
  },
  "amount": 100
}
```

Reply:
👍